shoes-swt/lib/shoes/swt/gradient.rb

```diff
@@ -44,13 +44,8 @@ def apply_as_stroke(gc, dsl)
       private
 
       def extract_dimensions(dsl)
-        if dsl.is_a?(Star)
-          left = dsl.element_left - dsl.element_width / 2.0
-          top  = dsl.element_top - dsl.element_height / 2.0
-        else
-          left = dsl.element_left
-          top  = dsl.element_top
-        end
+        left = dsl.element_left
+        top  = dsl.element_top
```

Comment:
yeah I guess it was from back in the days when each thing had its own dimensions and the one of star were slightly weird... at least that's what I tell myself when I hope I didn't break any important behavior!

Also good catch about unrolling, will do!